### PR TITLE
[4.0] Load webcomponents with asset manager

### PR DESCRIPTION
**Pull request by Fedik**
The pull depend from #25775, do not merge before that will be merged.

### Summary of Changes
This will load webcomponents with asset manager.

### Testing Instructions
Apply #25775 patch and apply current patch.
Click around, try create/edit content, all should work as before.

### Expected result
Everything works

### Actual result
Everything works

### Documentation Changes Required
Need to update Asset documentation to describe how to add webcomponent to it.

### CODE REVIEW

File: libraries/cms/html/email.php [hunk 1]
```diff
@@ -7,6 +7,7 @@
  * @license     GNU General Public License version 2 or later; see LICENSE.txt
  */
 
+use Joomla\CMS\Factory;
 use Joomla\CMS\HTML\HTMLHelper;
```
Comment (Quy) [inline libraries/cms/html/email.php]:
```suggestion
```

File: libraries/src/Document/Renderer/Html/MetasRenderer.php [hunk 2]
```diff
@@ -15,6 +15,7 @@
 use Joomla\CMS\Factory;
 use Joomla\CMS\Helper\TagsHelper;
 use Joomla\CMS\HTML\HTMLHelper;
```
Comment (Quy) [inline libraries/src/Document/Renderer/Html/MetasRenderer.php]:
```suggestion
```

File: libraries/src/Event/WebAsset/WebAssetRegistryAssetChanged.php [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+	 * @throws  BadMethodCallException
+	 *
+	 * @since  __DEPLOY_VERSION__
+	 */
+	public function __construct($name, array $arguments = array())
+	{
+		parent::__construct($name, $arguments);
+
+		// Check for required arguments
+		if (!array_key_exists('asset', $arguments) || !($arguments['asset'] instanceof WebAssetItemInterface))
```
Comment (Quy) [inline libraries/src/Event/WebAsset/WebAssetRegistryAssetChanged.php:39]:
```suggestion
		if (!\array_key_exists('asset', $arguments) || !($arguments['asset'] instanceof WebAssetItemInterface))
```

File: libraries/src/Event/WebAsset/WebAssetRegistryAssetChanged.php [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+		{
+			throw new BadMethodCallException("Argument 'asset' of event $name is not of the expected type");
+		}
+
+		if (!array_key_exists('assetType', $arguments) || !is_string($arguments['assetType']))
```
Comment (Quy) [inline libraries/src/Event/WebAsset/WebAssetRegistryAssetChanged.php:44]:
```suggestion
		if (!\array_key_exists('assetType', $arguments) || !is_string($arguments['assetType']))
```

File: libraries/src/Event/WebAsset/WebAssetRegistryAssetChanged.php [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+		{
+			throw new BadMethodCallException("Argument 'assetType' of event $name is not of the expected type");
+		}
+
+		if (!array_key_exists('change', $arguments) || !is_string($arguments['change']))
```
Comment (Quy) [inline libraries/src/Event/WebAsset/WebAssetRegistryAssetChanged.php:49]:
```suggestion
		if (!\array_key_exists('change', $arguments) || !is_string($arguments['change']))
```

File: libraries/src/Event/WebAsset/WebAssetRegistryAssetChanged.php (hunk 3 above)
Comment (Fedik) [reply libraries/src/Event/WebAsset/WebAssetRegistryAssetChanged.php:39]: this changes to  #25775

File: libraries/src/Event/WebAsset/WebAssetRegistryAssetChanged.php (hunk 5 above)
Comment (Fedik) [reply libraries/src/Event/WebAsset/WebAssetRegistryAssetChanged.php:49]: all under libraries/src/Event/WebAsset/ is to  #25775

File: administrator/components/com_associations/tmpl/association/edit.php [hunk 6]
```diff
@@ -22,7 +23,9 @@
 
 HTMLHelper::_('script', 'com_associations/sidebyside.js', ['version' => 'auto', 'relative' => true]);
 HTMLHelper::_('stylesheet', 'com_associations/sidebyside.css', ['version' => 'auto', 'relative' => true]);
-HTMLHelper::_('webcomponent', 'system/joomla-core-loader.min.js', ['version' => 'auto', 'relative' => true]);
+
+Factory::getDocument()->getWebAssetManager()
```
Comment (SharkyKZ) [inline administrator/components/com_associations/tmpl/association/edit.php]: Please use `$this->document` in component views.

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on 9de1195a3d2ddadbf2ff54568b86c8eca5352c47<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27268">issues.joomla.org/tracker/joomla-cms/27268</a>.</sub>

**Comment by Fedik** [comment]
I have redo #27418

**Pull request closed without merging**